src/GafferScene/Preview/RendererAlgo.cpp

```diff
@@ -157,9 +157,8 @@ class LocationTask : public tbb::task
 template <class Functor>
 void parallelProcessLocations( const GafferScene::ScenePlug *scene, Functor &f )
 {
-	Gaffer::ContextPtr c = new Gaffer::Context( *Gaffer::Context::current(), Gaffer::Context::Borrowed );
-	GafferScene::Filter::setInputScene( c.get(), scene );
-	LocationTask<Functor> *task = new( tbb::task::allocate_root() ) LocationTask<Functor>( scene, c.get(), ScenePlug::ScenePath(), f );
+	FilterPlug::SceneScope sceneScope( Gaffer::Context::current(), scene );
+	LocationTask<Functor> *task = new( tbb::task::allocate_root() ) LocationTask<Functor>( scene, Gaffer::Context::current(), ScenePlug::ScenePath(), f );
```

Comment:
I did think about that, but I wasn't sure. It felt a bit weird to get a cut-down just-setting-things view onto a temporary context and then have the option of getting the whole thing back again (albeit in const form), and I wasn't sure if it would get in the way of future optimisation. Then again, the whole thing appears again from `current()` anyway, so maybe it doesn't matter?